databases/week2/3School_query.sql

```diff
@@ -0,0 +1,22 @@
+USE School;
+CREATE TABLE `Class` (
+  `id` int(10) UNSIGNED NOT NULL AUTO_INCREMENT PRIMARY KEY,
+  `name` varchar(255) NOT NULL,
+  `begin_date` DATETIME NOT NULL,
+  `ends_date` DATETIME NOT NULL
+) ENGINE = InnoDB DEFAULT CHARSET = utf8mb4 COLLATE = utf8mb4_unicode_ci;
+CREATE TABLE `Student` (
+  `id` int(10) UNSIGNED NOT NULL AUTO_INCREMENT PRIMARY KEY,
+  `name` VARCHAR(255) NOT NULL,
+  `email` VARCHAR(255) NOT NULL,
+  `phone` VARCHAR(15),
+  `class_id` INT(10) UNSIGNED,
+  CONSTRAINT `fk_Class` FOREIGN KEY (`class_id`) REFERENCES `Class` (`id`) ON DELETE CASCADE ON UPDATE CASCADE
```

Comment:
when you delete a class you'll also delete the students that are part of it. is that what you want? :)

Reply:
Yes, you have three options for the ON DELETE: :
 - CASCADE: if you delete a class with id 2, all the students with a class_id=2 will also be deleted
 - SET NULL: if you delete a class with id 2, all the students with a class_id=2 will have NULL in the class_id (the class with id 2 no longer exists)
 - RESTRICT: you cannot delete classes with ids that are referenced in the student table
RESTRICT is the default, if you don't specify anything, that's what you get.
 
For "ON UPDATE", the behavior is the following:
 - CASCADE: if you change a class with id 2 to 4, all the students with a class_id=2 will have 4 in the class_id
 - SET NULL: if you change a class with id 2, all the students with a class_id=2 will have NULL in the class_id
 - RESTRICT: you cannot change the id in the class table, if that id referenced in the student table
RESTRICT is the default, if you don't specify anything, that's what you get.

Bottom line, SET NULL is generally a good idea for ON DELETE and CASCADE is generally a good idea for ON UPDATE.

Does it make more sense?